client/src/components/PropertiesList.tsx

```diff
@@ -62,7 +69,48 @@ const useValue = () => {
 
 export const FilterContext = React.createContext({} as ReturnType<typeof useValue>);
 
-const FilterContextProvider: React.FC<{}> = (props) => {
+export const filterAddrs = (addrs: AddressRecord[], filterSelections: FilterSelections) => {
+  const {
+    rsunitslatest: filterRsunitslatest,
+    ownernames: filterOwnernames,
+    unitsres: filterUnitsres,
+    zip: filterZip,
+  } = filterSelections;
+
+  return addrs.filter((addr) => {
+    let keepAddr = true;
+
+    if (keepAddr && filterRsunitslatest) {
+      keepAddr = (addr.rsunitslatest || 0) > 0;
+    }
+
+    if (keepAddr && !!filterOwnernames.length) {
+      const addrOwnernames =
+        addr.allcontacts &&
+        Object.entries(_groupBy(addr.allcontacts, "value")).map((contact) => contact[0]);
+
+      keepAddr = filterOwnernames.some((filterOwnername) =>
+        addrOwnernames?.includes(filterOwnername)
+      );
+    }
+
+    if (keepAddr && filterUnitsres.type !== "default") {
+      keepAddr = filterUnitsres.values.reduce(
+        (acc, rng) =>
```

Comment:
rng?

Reply:
range.. I was just copying some patterns in react-table functions, but yeah this isn't clear - fixing!